urllib3/util/retry.py

```diff
@@ -290,5 +290,31 @@ def __repr__(self):
                     cls=type(self), self=self)
 
 
+def handle_redirect(response, requester, retries, method, url, *args, **kwargs):
+    """
+    Abstraction of the redirect process so that we can access it from
+    multidomain levels in addition to from the :class:`.ConnectionPool`.
+    """
+    # urljoin will give us either a full URL or just a path, based on what
+    # the original URL is, making it appropriate for the ConnectionPool
+    # level as well as the PoolManager or SessionManager
+    redirect_location = urljoin(url, response.get_redirect_location())
+    if response.status == 303:
+        method = 'GET'
```

Comment:
I wouldn't use `_encode_url_methods`, that just defines methods with bodies. The definition for redirects is exclusive (i.e. everything but HEAD gets changed).

And no, you shouldn't be using `REDIRECT_STATUSES`, that would also be wrong. This is somewhat more complex than you'd expect. See [`requests.sessions.SessionRedirectMixin.rebuild_method`](https://github.com/kennethreitz/requests/blob/aa1c3adf9a2e5dcccd9c4d53b8d10c9dc4348f2f/requests/sessions.py#L249-L269) for the logic requests uses.